support/k8s/k8s-workload-registrar/controller.go

```diff
@@ -120,42 +122,28 @@ func (c *Controller) createPodEntry(ctx context.Context, pod *corev1.Pod) error
 		// has that annotation, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if annotationValue, ok := pod.Annotations[c.c.PodAnnotation]; ok {
-			return c.createPodEntryByAnnotation(ctx, pod, annotationValue)
+			spiffeID := c.makeID("%s", annotationValue)
+			return &spiffeID
 		}
 		return nil
 	}
 
 	// the controller has not been configured with a pod label or a pod annotation.
 	// create an entry based on the service account.
-	return c.createPodEntryByServiceAccount(ctx, pod)
+	spiffeID := c.makeID("ns/%s/sa/%s", pod.Namespace, pod.Spec.ServiceAccountName)
+	return &spiffeID
 }
 
-func (c *Controller) createPodEntryByLabel(ctx context.Context, pod *corev1.Pod, labelValue string) error {
-	return c.createEntry(ctx, &common.RegistrationEntry{
-		ParentId: c.nodeID(),
-		SpiffeId: c.makeID("%s", labelValue),
-		Selectors: []*common.Selector{
-			namespaceSelector(pod.Namespace),
-			podNameSelector(pod.Name),
-		},
-	})
-}
-
-func (c *Controller) createPodEntryByAnnotation(ctx context.Context, pod *corev1.Pod, annotationValue string) error {
-	return c.createEntry(ctx, &common.RegistrationEntry{
-		ParentId: c.nodeID(),
-		SpiffeId: c.makeID("%s", annotationValue),
-		Selectors: []*common.Selector{
-			namespaceSelector(pod.Namespace),
-			podNameSelector(pod.Name),
-		},
-	})
-}
+func (c *Controller) createPodEntry(ctx context.Context, pod *corev1.Pod) error {
+	spiffeID := c.podSpiffeID(pod)
```

Comment:
nit: unless you have to distinguish between "unset" and "empty", I find `*string` much less ergonomic than a `string` where `""` means unset.

Reply:
I suppose there's no reason to ever have an empty spiffe ID, so "" would work fine.  I think it'll clean up some of the tests a bit too.  Let me push that change